src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -339,37 +341,36 @@ void ProgFSO::arrangeFSC_and_fscGlobal(double sampling_rate,
 		}
 		mdRes.write(fnOut+"/GlobalFSC.xmd");
 
-		
-
-		// Here the FSC at 0.143 is obtained byt interpolating
-		FOR_ALL_ELEMENTS_IN_ARRAY1D(freq)
-		{
-			double ff = dAi(frc,i);
-			if ( (ff<=thrs) && (i>2) )
-			{
-				std::cout << std::endl << "shell = " << i << std::endl;
-				fscshellNum = i;
-				double y2, y1, x2, x1, slope, ny;
-				y2 = ff;
-				double ff_1 = dAi(freq, i-1);
-				y1 = ff_1;
-				x2 = ff;
-				x1 = ff_1;
-
-				slope = (y2 - y1)/(x2 - x1);
-				ny = y2 - slope*x2;
-
-				double fscResolution;
-				fscResolution = 1/(slope*thrs + ny);
-				std::cout << "Resolution " << fscResolution << std::endl;
-				break;
-			}
-		}
-
+		// Here the FSC at 0.143 is obtained by interpolating
+		fscInterpolation(freq, frc);
 
 		aniFilter.initZeros(freqidx);
+	}
 
+
+void ProgFSO::fscInterpolation(MultidimArray<double> &freq, MultidimArray< double > &frc)
```

Comment:
`const` freq and frc
Also, is it fine that you just report this resolution to standard output (and not e.g. save it for later)?